miner/src/block_assembler.rs

```diff
@@ -424,14 +429,41 @@ impl<CS: ChainStore + 'static> BlockAssembler<CS> {
 
         let witness = lock.into_witness();
         let input = CellInput::new_cellbase_input(candidate_number);
-        let output = CellOutput::new(block_reward, Bytes::default(), target_lock, None);
+
+        let r#type = None;
+
+        let data = self.custom_data(block_reward, &target_lock, &r#type)?;
+
+        let output = CellOutput::new(block_reward, data, target_lock, r#type);
+
         Ok(TransactionBuilder::default()
             .input(input)
             .output(output)
             .witness(witness)
             .build())
     }
 
+    fn custom_data(
```

Comment:
It's better to construct a cell output first with empty data, then the max allowed data length is

    capacity - occupied_capacity

Reply:
Better solution:

Construct the output first, truncate the tailing bytes if

```
occupied_capacity > capacity
```

Where `occupied_capacity - capacity` is the bytes to truncate.

If `occupied_capacity - capacity > data.length`, we should return an error.